conformance/lang/expressions/literals/floating_point_literal.balt

```diff
@@ -0,0 +1,5665 @@
+
+    c = 0XA.A;
+    io:println(c); // @output 10.625
+
+    c = 0XA.a;
+    io:println(c); // @output 10.625
+
+    c = 0Xa.A;
+    io:println(c); // @output 10.625
+
+    c = 0XaEbcD.0EaD;
+    io:println(c); // @output 715725.0573272705
+
+    c = 0XaE04bcD.0E91aD;
+    io:println(c); // @output 1.8247162905691034E8
+
+    c = 0X.0;
+    io:println(c); // @output 0.0
+
+    c = 0X.1;
+    io:println(c); // @output 0.0625
+
+    c = 0X.2345;
+    io:println(c); // @output 0.1377716064453125
+
+    c = 0X.A;
+    io:println(c); // @output 0.625
+
+    c = 0X.a;
+    io:println(c); // @output 0.625
+
+    c = 0x.aBdECFF;
+    io:println(c); // @output 0.671368595212698
+
+    c = 0x.aBd0E12C9FF4;
+    io:println(c); // @output 0.6711559995600993
+
+    c = 0X.2p+0;
+    io:println(c); // @output 0.125
+
+    c = 0X.AP-0;
+    io:println(c); // @output 0.625
+
+    c = 0X.ap-0;
+    io:println(c); // @output 0.625
+
+    c = 0x.263567p+46;
+    io:println(c); // @output 1.0502700793856E13
+
+    c = 0X.AEDP-46;
+    io:println(c); // @output 9.704043124614259E-15
+
+    c = 0X.aedp-46;
+    io:println(c); // @output 9.704043124614259E-15
+
+    c = 0x.1BeAP+948;
+    io:println(c); // @output 2.594292682612851E284
+
+    c = 0X.0EfCap-456;
+    io:println(c); // @output 3.1462367457512917E-139
+
+    c = 0x0.0p-0;
+    io:println(c); // @output 0.0
+
+    c = 0x0.0P+0;
+    io:println(c); // @output 0.0
+
+    c = 0x9.4p-1;
+    io:println(c); // @output 4.625
+
+    c = 0x9546.4748p+126;
+    io:println(c); // @output 3.250911279637454E42
+
+    c = 0xa.aP-1;
+    io:println(c); // @output 5.3125
+
+    c = 0Xa.ap+1;
+    io:println(c); // @output 21.25
+
+    c = 0xA.aP-12;
+    io:println(c); // @output 0.002593994140625
+
+    c = 0XA.ap+12;
+    io:println(c); // @output 43520.0
+
+    c = 0xaeb.acdfP-302;
+    io:println(c); // @output 3.431057809225094E-88
+
+    c = 0xABC.CEEFFp+302;
+    io:println(c); // @output 2.2397685993041007E94
+
+    c = 0xAbC.0eEfFp-453;
+    io:println(c); // @output 1.1815113934486016E-133
+
+    c = 0xAb12C.0e9E1f0FP+45;
+    io:println(c); // @output 2.465425448163409E19
+}
+
+Test-Case: output
+Description: Test floating point literal max and min with float type descriptor.
+Labels: float, HexFloatingPointLiteral
+
+function init() {
+
+    float d = 1.7976931348623157E+308;
+    io:println(d); // @output 1.7976931348623157E308
+
+    d = 4.94065645841246544E-324;
+    io:println(d); // @output 4.9E-324
+
+    d = 0x1.0fffffffffffffp+1023;
+    io:println(d); // @output 9.550244778956053E307
+
+    d = 0x1p-1074;
+    io:println(d); // @output 4.9E-324
+
+    d = 0x0p-1074;
+    io:println(d); // @output 0.0
+
+    d = 0x0p-1075;
+    io:println(d); // @output 0.0
+}
+
+Test-Case: output
+Description: Test floating point literal max and min with decimal type descriptor.
+Labels: decimal
+
+function init() {
+    decimal e = 1E-6176;
+    io:println(e); // @output 0
+
+    e = 9.99E+6111;
```

Comment:
We can add the largest negative value.

Also correct values are,
https://github.com/ballerina-platform/ballerina-lang/blob/747be34aa8ac3e494500790e134add9ccdab4769/bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/DecimalValue.java#L52